wallet-new/src/Cardano/Wallet/Kernel/DB/Spec/Pending.hs

```diff
@@ -197,6 +200,20 @@ liftMap2 :: (UnderlyingMap -> UnderlyingMap -> UnderlyingMap)
          -> Pending -> Pending -> Pending
 liftMap2 f (toMap -> p) (toMap -> p') = fromMap (f p p')
 
+{--------------------------------------------------------------------------------
+  Compression
+-------------------------------------------------------------------------------}
+
+type PendingDiff = InDb (Map Core.TxId Core.TxAux, Set Core.TxId)
+
+liftDeltaPending :: (UnderlyingMap -> UnderlyingMap -> (Map Core.TxId Core.TxAux, Set Core.TxId))
```

Comment:
Sorry for being a pain in the neck, but I do wonder -- how do we feel about the fact this function requires the caller to pass a function which works on an internal type, of this module, i.e. the `UnderlyingMap`?

Reply:
Yes, not the best I guess. I hope I got it right now.